crates/aptos-crypto/src/unit_tests/p256_test.rs

```diff
@@ -0,0 +1,204 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+#![allow(clippy::redundant_clone)] // Required to work around prop_assert_eq! limitations
+
+use crate as aptos_crypto;
+use crate::{
+    p256::{P256_SIGNATURE_LENGTH, P256PublicKey, P256_PUBLIC_KEY_LENGTH, P256_PRIVATE_KEY_LENGTH, P256PrivateKey, P256Signature, ORDER_HALF}, 
+    test_utils::{
+        random_serializable_struct,
+        uniform_keypair_strategy,
+    },
+    traits::*,
+};
+use signature::Verifier;
+use p256::{EncodedPoint, NonZeroScalar};
+use aptos_crypto_derive::{BCSCryptoHash, CryptoHasher};
+use core::convert::TryFrom;
+use proptest::{collection::vec, prelude::*};
+use serde::{Deserialize, Serialize};
+
+#[derive(CryptoHasher, BCSCryptoHash, Serialize, Deserialize)]
+struct CryptoHashable(pub usize);
+proptest! {
+    #[test]
+    fn test_pub_key_deserialization(bits in any::<[u8; 32]>()){
+        let pt_deser = EncodedPoint::from_bytes(&bits[..]);
+        let pub_key = P256PublicKey::try_from(&bits[..]);
+        let check = matches!((pt_deser, pub_key),
+            (Ok(_), Ok(_)) // we agree with RustCrypto's sec1 implementation,
+            | (Err(_), Err(_)) // we agree on point decompression failures,
+        );
+        prop_assert!(check);
+    }
+
+    #[test]
+    fn test_keys_encode(keypair in uniform_keypair_strategy::<P256PrivateKey, P256PublicKey>()) {
+        {
+            let encoded = keypair.private_key.to_encoded_string().unwrap();
+            // Hex encoding of a 64-bytes key is 128 (2 x 64) characters.
+            prop_assert_eq!(2 + 2 * P256_PRIVATE_KEY_LENGTH, encoded.len());
```

Comment:
Where are the extra 2 bytes from?

Reply:
It's for the `0x` prepended to the hex encoding. Changed the comment to explain this